cpp/tests/scalar/factories_test.cpp

```diff
@@ -161,4 +163,35 @@ TYPED_TEST(FixedPointScalarFactory, ValueProvided)
   EXPECT_TRUE(s->is_valid());
 }
 
+struct StructScalarFactory : public ScalarFactoryTest {
+};
+
+TEST_F(StructScalarFactory, Basic)
+{
+  cudf::test::fixed_width_column_wrapper<int> col0{1};
+  cudf::test::strings_column_wrapper col1{"abc"};
+  cudf::test::lists_column_wrapper<int> col2{{1, 2, 3}};
+  cudf::test::structs_column_wrapper struct_col({col0, col1, col2});
+  cudf::column_view cv = static_cast<cudf::column_view>(struct_col);
+  std::vector<cudf::column_view> children(cv.child_begin(), cv.child_end());
+
+  // table_view constructor
+  {
+    auto sc    = cudf::make_struct_scalar(cudf::table_view{children});
+    auto s     = static_cast<cudf::scalar_type_t<cudf::struct_view>*>(sc.get());
+    auto sview = s->view();
+    EXPECT_TRUE(s->is_valid());
+    CUDF_TEST_EXPECT_TABLES_EQUIVALENT(cudf::table_view{children}, s->view());
+  }
+
+  // host_span constructor
+  {
+    auto sc    = cudf::make_struct_scalar(cudf::host_span<cudf::column_view const>{children});
```

Comment:
same here, not sure if the explicit instantiation has been purposely added.

Reply:
Same as above.